server/src/Commands.def

```diff
@@ -12,7 +12,9 @@
 #error Please define the 'X(id, command, description)' macro before inclusion!
 #endif
 
-X(Candidates, "candidates", "print completion candidates (require previous complete)")
+X(Candidates, "candidates",
+  "PREFIX STYLE - print completion candidates (require previous complete). "
```

Comment:
> So need we create a new variable in elisp or just reuse company-irony-ignore-case?

For company-irony company-irony-ignore-case is good.
For irony-completion, change the following signature:

```el
(defun irony-completion-candidates (&optional prefix ignore-case)
```

to:

```el
(defun irony-completion-candidates (&optional prefix style)
```


> So we need create a new enum like 'enum Style' (or other name if you have better option :)) and a new StyleConverter to validate it

- `enum class Style`
- `enum class PrefixStyle`
- `enum class MatchStyle`
- `enum class PrefixMatchStyle`
- `enum class PrefixMatchStyle`
- `enum class CandidatesStyle`

My favorite is `PrefixMatchStyle`, but you can pick up anyone you want, or one of your own if you find something better.

And yes a new converter, you are right.


> Both irony.cpp and command.cpp need access this enum. Is it ok to create a new header file for this enum?

New header is fine. If I had to pick an existing one, I would choose Irony.h, but new header is good to me too.